storage/innobase/page/page0page.cc

```diff
@@ -627,8 +627,7 @@ page_copy_rec_list_end(
 			have at least one predecessor (the predefined
 			infimum record, or a freshly copied record
 			that is smaller than "ret"). */
-			if (UNIV_UNLIKELY(!ret_pos
-					  || ret_pos == ULINT_UNDEFINED)) {
+			if (UNIV_UNLIKELY(ret_pos == ULINT_UNDEFINED)) {
```

Comment:
The comment still mentions `ret_pos >=1` even though the condition was removed. When would the successor of the infimum record be 0?

Is this change really needed? The condition may look somewhat redundant, but its purpose is to catch corruption. The `ret = page_rec_get_next(page_get_infimum_rec(new_page))` and `page_rec_get_n_recs_before(ret)` navigate different parts of the page (the next-record list and the sparse page directory).